crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -695,79 +697,20 @@ impl<T: WasmInstance> WasmModuleInstance<T> {
-                    col_name: ty.name.clone().context("column without name")?,
-                    col_type: ty.algebraic_type.clone(),
-                    is_autoinc: attr.is_autoinc(),
-                })
-            })
-            .collect::<anyhow::Result<_>>()?;
-
-        let mut indexes = Vec::new();
-        for (col_id, col) in columns.iter().enumerate() {
-            let mut index_for_column = None;
-            for index in table.indexes.iter() {
-                let [index_col_id] = *index.col_ids else {
-                    //Ignore multi-column indexes
-                    continue;
-                };
-                if index_col_id as usize != col_id {
-                    continue;
-                }
-                index_for_column = Some(index);
-                break;
-            }
-
-            let col_attr = table.column_attrs.get(col_id).context("invalid column id")?;
-            // If there's an index defined for this column already, use it
-            // making sure that it is unique if the column has a unique constraint
-            if let Some(index) = index_for_column {
-                match index.ty {
-                    IndexType::BTree => {}
-                    // TODO
-                    IndexType::Hash => anyhow::bail!("hash indexes not yet supported"),
-                }
-                let index = IndexDef {
-                    table_id: 0, // Will be ignored
-                    cols: NonEmpty::new(col_id as u32),
-                    name: index.name.clone(),
-                    is_unique: col_attr.is_unique(),
-                };
-                indexes.push(index);
-            } else if col_attr.is_unique() {
-                // If you didn't find an index, but the column is unique then create a unique btree index
-                // anyway.
-                let index = IndexDef {
-                    table_id: 0, // Will be ignored
-                    cols: NonEmpty::new(col_id as u32),
-                    name: format!("{}_{}_unique", table.name, col.col_name),
-                    is_unique: true,
-                };
-                indexes.push(index);
-            }
-        }
-
-        Ok(TableDef {
-            table_name: table.name.clone(),
-            columns,
-            indexes,
-            table_type: table.table_type,
-            table_access: table.table_access,
-        })
+    fn schema_for(&self, table: &spacetimedb_lib::TableDesc) -> anyhow::Result<TableDef> {
+        // let schema = self
+        //     .info
+        //     .typespace
+        //     .with_type(&table.data)
+        //     .resolve_refs()
+        //     .context("recursive types not yet supported")?;
+        // let schema = schema.into_product().ok().context("table not a product type?")?;
+        // anyhow::ensure!(
+        //     table.schema.columns.len() == schema.elements.len(),
+        //     "mismatched number of columns"
+        // );
+
+        Ok(table.schema.clone())
```

Comment:
If this is no longer necessary, remove the method?

Reply:
I was waiting for confirmation :)